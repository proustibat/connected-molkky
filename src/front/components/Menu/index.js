import React, { useEffect } from 'react';
import MenuLinks from '@components/Menu/MenuLinks';
import PropTypes from 'prop-types';

const Menu = ({ title }) => {
  useEffect(() => {
    // Mobile menu initialization
    const { M: MaterializeCSS } = window;
    MaterializeCSS && MaterializeCSS.Sidenav.init(document.querySelectorAll('.sidenav'), { edge: 'right' });
  }, []);

  return (
    <div className="navbar-fixed">
      <nav>
        <div className="container nav-wrapper">
          <a href="/molkky/game" className="brand-logo left">{title}</a>
          <a href="/molkky/game" data-target="mobile-menu" className="sidenav-trigger right"><i className="material-icons">menu</i></a>
          <MenuLinks className="right hide-on-med-and-down" />
        </div>
      </nav>
      <MenuLinks id="mobile-menu" className="sidenav right" />
    </div>
  );
};

Menu.propTypes = {
  title: PropTypes.string.isRequired,
};

export default Menu;
