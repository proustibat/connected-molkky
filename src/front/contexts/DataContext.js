import React, { useContext } from 'react';

export const DataContext = React.createContext([]);
export const useDataContext = () => useContext(DataContext);
export const DataContextProvider = DataContext.Provider;
export const DataContextConsumer = DataContext.Consumer;

export default DataContext;
