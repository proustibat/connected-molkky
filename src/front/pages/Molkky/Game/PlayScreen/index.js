import React from 'react';

const PlayScreen = () => (
  <h1>PlayScreen</h1>
);

export default PlayScreen;
