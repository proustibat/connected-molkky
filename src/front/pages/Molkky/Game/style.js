export default {
  loadingBar: {
    height: 'auto',
    position: 'absolute',
    top: 56,
  },
  loadingBarProgress: {
    margin: 0,
  },
};
