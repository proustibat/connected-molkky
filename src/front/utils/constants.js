export default {
  POSITION: {
    KNOCKED_OVER: 'KNOCKED_OVER',
    UPRIGHT: 'UPRIGHT',
  },

  MESSAGE_TYPE: {
    INFO: 'info',
    WARNING: 'warning',
    ERROR: 'error',
  },
};
