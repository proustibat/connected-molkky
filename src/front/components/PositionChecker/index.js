import React from 'react';
import kebabCase from 'lodash/kebabCase';
import SkittlesDisplay from '@components/SkittlesDisplay';
import { useDataContext } from '@contexts/DataContext';
import constants from '@utils/constants';
import Button from '@components/Button';
import MessageIcon from '@components/MessageIcon';

const PositionChecker = () => {
  const data = useDataContext();

  const onStartClick = () => {
    // TODO
  };

  const renderWarning = (message, positionType) => data
    .some(({ position }) => position === positionType) && (
    <>
      <MessageIcon type={constants.MESSAGE_TYPE.WARNING}>
        {message}
      </MessageIcon>
      <ul className="collection">
        {
          data
            .filter(({ position }) => position === positionType)
            .map(({ value }) => <li key={`${kebabCase(positionType || 'missing')}-${value}`} className="collection-item">{value}</li>)
        }
      </ul>
    </>
  );

  const renderError = () => data.length === 0 && (
    <MessageIcon type={constants.MESSAGE_TYPE.ERROR}>
      Oops! Please move the pins to detect them!
    </MessageIcon>
  );

  return (
    <>
      <p className="flow-text">Be sure pins are placed as follows then start the game</p>
      <SkittlesDisplay />
      <Button
        disabled={
          data.length === 0
          || !(data.every(({ position }) => position === constants.POSITION.UPRIGHT))
        }
        style={{ marginBottom: '1rem' }}
        onClick={onStartClick}
      >
        Start
      </Button>
      {renderWarning('Please stand up the following pins:', constants.POSITION.KNOCKED_OVER)}
      {renderWarning('All pins are not detected, try to move the missing pins:', null)}
      {renderError()}
    </>
  );
};

export default PositionChecker;
