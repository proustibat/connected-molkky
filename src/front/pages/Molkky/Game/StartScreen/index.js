import { Link, useHistory } from 'react-router-dom';
import React, { useState } from 'react';
import { resetGame, startGame } from '@root/front/services/api';
import Button from '@components/Button';
import PositionChecker from '@components/PositionChecker';
import TeamButton from '@components/TeamButton';
import { useDataContext } from '@contexts/DataContext';
import { usePlayContext } from '@contexts/PlayContext';

const StartScreen = () => {
  const history = useHistory();

  const {
    teams, currentTurn, setCurrentTurn, scores, setScores,
  } = usePlayContext();
  const { setIsLoading } = useDataContext();

  const [startReady, setStartReady] = useState(false);
  const [selectedTeam, setSelectedTeam] = useState(Object.entries(teams)[0][0]);

  const onTeamSelect = (team) => {
    setSelectedTeam(team);
  };

  const onStartClick = (restart = false) => async () => {
    setIsLoading(true);
    const result = restart
      ? await resetGame()
      : await startGame({ teams: Object.keys(teams), playingTeam: selectedTeam });
    setIsLoading(false);
    if (result) {
      setCurrentTurn(result.currentTurn);
      setScores(result.scores);
      history.push('/molkky/game/play');
    }
  };

  return (!currentTurn && !scores) ? (
    <>
      <div className="section">
        <div className="container">
          <h1>Players</h1>
          <p className="flow-text">Which team plays first?</p>
          <div style={{ display: 'flex', justifyContent: 'center' }}>
            {
              Object.entries(teams).map(([value, { icon, name }], i) => (
                <TeamButton
                  key={value}
                  icon={icon}
                  style={(i % 2 === 0) ? { marginRight: '2rem' } : { marginLeft: '2rem' }}
                  name={name}
                  value={value}
                  onClick={onTeamSelect}
                  selected={selectedTeam === value}
                />
              ))
            }
          </div>
        </div>
      </div>
      <div className="divider" />
      <div className="section">
        <div className="container">
          <h1>Playground</h1>
          <PositionChecker onReadyChange={setStartReady} />
          <Button onClick={onStartClick()} disabled={!startReady}>Play</Button>
        </div>
      </div>
    </>
  ) : (
    <div className="section">
      <div className="container">
        <h1>A game has already been started!</h1>
        <p>Do you want to continue the same or restart a new one ?</p>
        <Link className="waves-effect waves-light z-depth-1 btn" to="/molkky/game/play">Continue the current game</Link>
        <Button onClick={onStartClick(true)}>Start a new game</Button>
      </div>
    </div>
  );
};

export default StartScreen;
