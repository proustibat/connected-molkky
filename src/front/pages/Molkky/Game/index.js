import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import PositionChecker from '@components/PositionChecker';
import { DataContextProvider } from '@root/front/contexts/DataContext';
import { getRandomPosition } from '@utils/services';
import Menu from '@components/Menu';

const Game = ({ title }) => {
  const [positionData, setPositionData] = useState([]);

  useEffect(() => {
    // Get random data for skittles
    setPositionData(Array.from({ length: 12 }, (_, i) => ({
      value: i + 1,
      position: getRandomPosition(),
    })));
  }, []);

  return (
    <DataContextProvider value={positionData}>
      <Menu title={title} />
      <div className="section no-pad-bot">
        <div className="container">
          <PositionChecker />
        </div>
      </div>
    </DataContextProvider>
  );
};

Game.propTypes = {
  title: PropTypes.string,
};

Game.defaultProps = {
  title: null,
};

Game.displayName = 'Game';

export default Game;
