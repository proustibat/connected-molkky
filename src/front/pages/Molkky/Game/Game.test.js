import React from 'react';
import { shallow, mount } from 'enzyme';
import { act } from 'react-dom/test-utils';
import { DataContextProvider } from '@contexts/DataContext';
import Menu from '@components/Menu';
import PositionChecker from '@components/PositionChecker';
import * as services from '@utils/services';
import constants from '@utils/constants';

import Game from './index';

const givenProps = {
  title: 'Game',
};

describe('Game', () => {
  beforeAll(() => {
    jest.spyOn(services, 'getRandomPosition').mockReturnValue(constants.POSITION.UPRIGHT);
  });

  afterEach(() => {
    services.getRandomPosition.mockClear();
  });

  afterAll(() => {
    services.getRandomPosition.mockRestore();
  });

  it('should render the component correctly', () => {
    // Given / When
    const component = shallow(<Game {...givenProps} />);

    // Then
    expect(component).toHaveLength(1);
    expect(component.find(DataContextProvider)).toHaveLength(1);
    expect(component.find(Menu)).toHaveLength(1);
    expect(component.find(PositionChecker)).toHaveLength(1);
  });

  it('should get random data on mount', () => {
    // Given / When
    let component;
    act(() => { component = mount(<Game {...givenProps} />); });

    // Then
    expect(component).toMatchSnapshot();
  });
});
