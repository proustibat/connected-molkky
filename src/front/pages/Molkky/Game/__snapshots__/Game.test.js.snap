// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game should get random data on mount 1`] = `
<Game
  title="Game"
>
  <Menu
    title="Game"
  >
    <div
      className="navbar-fixed"
    >
      <nav>
        <div
          className="container nav-wrapper"
        >
          <a
            className="brand-logo left"
            href="/"
          >
            Game
          </a>
          <a
            className="sidenav-trigger right"
            data-target="mobile-menu"
            href="/"
          >
            <i
              className="material-icons"
            >
              menu
            </i>
          </a>
          <MenuLinks
            className="right hide-on-med-and-down"
            id=""
          >
            <ul
              className="right hide-on-med-and-down"
              id=""
            >
              <li>
                <a
                  href="/"
                >
                  Link 1
                </a>
              </li>
              <li>
                <a
                  href="/"
                >
                  Link 2
                </a>
              </li>
            </ul>
          </MenuLinks>
        </div>
      </nav>
      <MenuLinks
        className="sidenav right"
        id="mobile-menu"
      >
        <ul
          className="sidenav right"
          id="mobile-menu"
        >
          <li>
            <a
              href="/"
            >
              Link 1
            </a>
          </li>
          <li>
            <a
              href="/"
            >
              Link 2
            </a>
          </li>
        </ul>
      </MenuLinks>
    </div>
  </Menu>
  <div
    className="section no-pad-bot"
  >
    <div
      className="container"
    >
      <PositionChecker>
        <p
          className="flow-text"
        >
          Be sure pins are placed as follows then start the game
        </p>
        <SkittlesDisplay>
          <div
            style={
              Object {
                "display": "flex",
                "flexDirection": "column",
                "marginBottom": "2rem",
              }
            }
          >
            <div
              style={
                Object {
                  "display": "flex",
                  "justifyContent": "center",
                }
              }
            >
              <Skittle
                position={null}
                value={7}
              >
                <SVGCircle
                  fillColor="#d7ffff"
                  height={70}
                  strokeColor="#0944ff"
                  strokeWidth={25}
                  text={7}
                  textColor="#0944ff"
                  textStyle={
                    Object {
                      "fontSize": 175,
                    }
                  }
                  width={70}
                >
                  <svg
                    height={70}
                    viewBox="-25 -25 400 400"
                    width={70}
                  >
                    <circle
                      cx="175"
                      cy="175"
                      fill="#d7ffff"
                      r="175"
                      stroke="#0944ff"
                      strokeWidth={25}
                    />
                    <text
                      dominantBaseline="central"
                      fill="#0944ff"
                      style={
                        Object {
                          "fontSize": 175,
                        }
                      }
                      textAnchor="middle"
                      x={175}
                      y={175}
                    >
                      7
                    </text>
                  </svg>
                </SVGCircle>
              </Skittle>
              <Skittle
                position={null}
                value={8}
              >
                <SVGCircle
                  fillColor="#d7ffff"
                  height={70}
                  strokeColor="#0944ff"
                  strokeWidth={25}
                  text={8}
                  textColor="#0944ff"
                  textStyle={
                    Object {
                      "fontSize": 175,
                    }
                  }
                  width={70}
                >
                  <svg
                    height={70}
                    viewBox="-25 -25 400 400"
                    width={70}
                  >
                    <circle
                      cx="175"
                      cy="175"
                      fill="#d7ffff"
                      r="175"
                      stroke="#0944ff"
                      strokeWidth={25}
                    />
                    <text
                      dominantBaseline="central"
                      fill="#0944ff"
                      style={
                        Object {
                          "fontSize": 175,
                        }
                      }
                      textAnchor="middle"
                      x={175}
                      y={175}
                    >
                      8
                    </text>
                  </svg>
                </SVGCircle>
              </Skittle>
              <Skittle
                position={null}
                value={9}
              >
                <SVGCircle
                  fillColor="#d7ffff"
                  height={70}
                  strokeColor="#0944ff"
                  strokeWidth={25}
                  text={9}
                  textColor="#0944ff"
                  textStyle={
                    Object {
                      "fontSize": 175,
                    }
                  }
                  width={70}
                >
                  <svg
                    height={70}
                    viewBox="-25 -25 400 400"
                    width={70}
                  >
                    <circle
                      cx="175"
                      cy="175"
                      fill="#d7ffff"
                      r="175"
                      stroke="#0944ff"
                      strokeWidth={25}
                    />
                    <text
                      dominantBaseline="central"
                      fill="#0944ff"
                      style={
                        Object {
                          "fontSize": 175,
                        }
                      }
                      textAnchor="middle"
                      x={175}
                      y={175}
                    >
                      9
                    </text>
                  </svg>
                </SVGCircle>
              </Skittle>
            </div>
            <div
              style={
                Object {
                  "display": "flex",
                  "justifyContent": "center",
                }
              }
            >
              <Skittle
                position={null}
                value={5}
              >
                <SVGCircle
                  fillColor="#d7ffff"
                  height={70}
                  strokeColor="#0944ff"
                  strokeWidth={25}
                  text={5}
                  textColor="#0944ff"
                  textStyle={
                    Object {
                      "fontSize": 175,
                    }
                  }
                  width={70}
                >
                  <svg
                    height={70}
                    viewBox="-25 -25 400 400"
                    width={70}
                  >
                    <circle
                      cx="175"
                      cy="175"
                      fill="#d7ffff"
                      r="175"
                      stroke="#0944ff"
                      strokeWidth={25}
                    />
                    <text
                      dominantBaseline="central"
                      fill="#0944ff"
                      style={
                        Object {
                          "fontSize": 175,
                        }
                      }
                      textAnchor="middle"
                      x={175}
                      y={175}
                    >
                      5
                    </text>
                  </svg>
                </SVGCircle>
              </Skittle>
              <Skittle
                position={null}
                value={11}
              >
                <SVGCircle
                  fillColor="#d7ffff"
                  height={70}
                  strokeColor="#0944ff"
                  strokeWidth={25}
                  text={11}
                  textColor="#0944ff"
                  textStyle={
                    Object {
                      "fontSize": 175,
                    }
                  }
                  width={70}
                >
                  <svg
                    height={70}
                    viewBox="-25 -25 400 400"
                    width={70}
                  >
                    <circle
                      cx="175"
                      cy="175"
                      fill="#d7ffff"
                      r="175"
                      stroke="#0944ff"
                      strokeWidth={25}
                    />
                    <text
                      dominantBaseline="central"
                      fill="#0944ff"
                      style={
                        Object {
                          "fontSize": 175,
                        }
                      }
                      textAnchor="middle"
                      x={175}
                      y={175}
                    >
                      11
                    </text>
                  </svg>
                </SVGCircle>
              </Skittle>
              <Skittle
                position={null}
                value={12}
              >
                <SVGCircle
                  fillColor="#d7ffff"
                  height={70}
                  strokeColor="#0944ff"
                  strokeWidth={25}
                  text={12}
                  textColor="#0944ff"
                  textStyle={
                    Object {
                      "fontSize": 175,
                    }
                  }
                  width={70}
                >
                  <svg
                    height={70}
                    viewBox="-25 -25 400 400"
                    width={70}
                  >
                    <circle
                      cx="175"
                      cy="175"
                      fill="#d7ffff"
                      r="175"
                      stroke="#0944ff"
                      strokeWidth={25}
                    />
                    <text
                      dominantBaseline="central"
                      fill="#0944ff"
                      style={
                        Object {
                          "fontSize": 175,
                        }
                      }
                      textAnchor="middle"
                      x={175}
                      y={175}
                    >
                      12
                    </text>
                  </svg>
                </SVGCircle>
              </Skittle>
              <Skittle
                position={null}
                value={6}
              >
                <SVGCircle
                  fillColor="#d7ffff"
                  height={70}
                  strokeColor="#0944ff"
                  strokeWidth={25}
                  text={6}
                  textColor="#0944ff"
                  textStyle={
                    Object {
                      "fontSize": 175,
                    }
                  }
                  width={70}
                >
                  <svg
                    height={70}
                    viewBox="-25 -25 400 400"
                    width={70}
                  >
                    <circle
                      cx="175"
                      cy="175"
                      fill="#d7ffff"
                      r="175"
                      stroke="#0944ff"
                      strokeWidth={25}
                    />
                    <text
                      dominantBaseline="central"
                      fill="#0944ff"
                      style={
                        Object {
                          "fontSize": 175,
                        }
                      }
                      textAnchor="middle"
                      x={175}
                      y={175}
                    >
                      6
                    </text>
                  </svg>
                </SVGCircle>
              </Skittle>
            </div>
            <div
              style={
                Object {
                  "display": "flex",
                  "justifyContent": "center",
                }
              }
            >
              <Skittle
                position={null}
                value={3}
              >
                <SVGCircle
                  fillColor="#d7ffff"
                  height={70}
                  strokeColor="#0944ff"
                  strokeWidth={25}
                  text={3}
                  textColor="#0944ff"
                  textStyle={
                    Object {
                      "fontSize": 175,
                    }
                  }
                  width={70}
                >
                  <svg
                    height={70}
                    viewBox="-25 -25 400 400"
                    width={70}
                  >
                    <circle
                      cx="175"
                      cy="175"
                      fill="#d7ffff"
                      r="175"
                      stroke="#0944ff"
                      strokeWidth={25}
                    />
                    <text
                      dominantBaseline="central"
                      fill="#0944ff"
                      style={
                        Object {
                          "fontSize": 175,
                        }
                      }
                      textAnchor="middle"
                      x={175}
                      y={175}
                    >
                      3
                    </text>
                  </svg>
                </SVGCircle>
              </Skittle>
              <Skittle
                position={null}
                value={10}
              >
                <SVGCircle
                  fillColor="#d7ffff"
                  height={70}
                  strokeColor="#0944ff"
                  strokeWidth={25}
                  text={10}
                  textColor="#0944ff"
                  textStyle={
                    Object {
                      "fontSize": 175,
                    }
                  }
                  width={70}
                >
                  <svg
                    height={70}
                    viewBox="-25 -25 400 400"
                    width={70}
                  >
                    <circle
                      cx="175"
                      cy="175"
                      fill="#d7ffff"
                      r="175"
                      stroke="#0944ff"
                      strokeWidth={25}
                    />
                    <text
                      dominantBaseline="central"
                      fill="#0944ff"
                      style={
                        Object {
                          "fontSize": 175,
                        }
                      }
                      textAnchor="middle"
                      x={175}
                      y={175}
                    >
                      10
                    </text>
                  </svg>
                </SVGCircle>
              </Skittle>
              <Skittle
                position={null}
                value={4}
              >
                <SVGCircle
                  fillColor="#d7ffff"
                  height={70}
                  strokeColor="#0944ff"
                  strokeWidth={25}
                  text={4}
                  textColor="#0944ff"
                  textStyle={
                    Object {
                      "fontSize": 175,
                    }
                  }
                  width={70}
                >
                  <svg
                    height={70}
                    viewBox="-25 -25 400 400"
                    width={70}
                  >
                    <circle
                      cx="175"
                      cy="175"
                      fill="#d7ffff"
                      r="175"
                      stroke="#0944ff"
                      strokeWidth={25}
                    />
                    <text
                      dominantBaseline="central"
                      fill="#0944ff"
                      style={
                        Object {
                          "fontSize": 175,
                        }
                      }
                      textAnchor="middle"
                      x={175}
                      y={175}
                    >
                      4
                    </text>
                  </svg>
                </SVGCircle>
              </Skittle>
            </div>
            <div
              style={
                Object {
                  "display": "flex",
                  "justifyContent": "center",
                }
              }
            >
              <Skittle
                position={null}
                value={1}
              >
                <SVGCircle
                  fillColor="#d7ffff"
                  height={70}
                  strokeColor="#0944ff"
                  strokeWidth={25}
                  text={1}
                  textColor="#0944ff"
                  textStyle={
                    Object {
                      "fontSize": 175,
                    }
                  }
                  width={70}
                >
                  <svg
                    height={70}
                    viewBox="-25 -25 400 400"
                    width={70}
                  >
                    <circle
                      cx="175"
                      cy="175"
                      fill="#d7ffff"
                      r="175"
                      stroke="#0944ff"
                      strokeWidth={25}
                    />
                    <text
                      dominantBaseline="central"
                      fill="#0944ff"
                      style={
                        Object {
                          "fontSize": 175,
                        }
                      }
                      textAnchor="middle"
                      x={175}
                      y={175}
                    >
                      1
                    </text>
                  </svg>
                </SVGCircle>
              </Skittle>
              <Skittle
                position={null}
                value={2}
              >
                <SVGCircle
                  fillColor="#d7ffff"
                  height={70}
                  strokeColor="#0944ff"
                  strokeWidth={25}
                  text={2}
                  textColor="#0944ff"
                  textStyle={
                    Object {
                      "fontSize": 175,
                    }
                  }
                  width={70}
                >
                  <svg
                    height={70}
                    viewBox="-25 -25 400 400"
                    width={70}
                  >
                    <circle
                      cx="175"
                      cy="175"
                      fill="#d7ffff"
                      r="175"
                      stroke="#0944ff"
                      strokeWidth={25}
                    />
                    <text
                      dominantBaseline="central"
                      fill="#0944ff"
                      style={
                        Object {
                          "fontSize": 175,
                        }
                      }
                      textAnchor="middle"
                      x={175}
                      y={175}
                    >
                      2
                    </text>
                  </svg>
                </SVGCircle>
              </Skittle>
            </div>
          </div>
        </SkittlesDisplay>
        <Button
          disabled={true}
          onClick={[Function]}
          size="btn-large"
          style={
            Object {
              "marginBottom": "1rem",
            }
          }
        >
          <button
            className="waves-effect waves-light z-depth-2 btn btn-large disabled"
            onClick={[Function]}
            style={
              Object {
                "marginBottom": "1rem",
              }
            }
            type="button"
          >
            Start
          </button>
        </Button>
        <MessageIcon
          type="error"
        >
          <p
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
              }
            }
          >
            <i
              className="small material-icons"
              style={
                Object {
                  "color": "#b71c1c",
                }
              }
            >
              error
            </i>
            <span
              style={
                Object {
                  "borderBottom": "1px solid #b71c1c",
                  "lineHeight": "1rem",
                }
              }
            >
              Oops! Please move the pins to detect them!
            </span>
          </p>
        </MessageIcon>
      </PositionChecker>
    </div>
  </div>
</Game>
`;
